### DISCUSSION TITLE
Clickhouse migration error

### DISCUSSION THREAD

**Discussion by nicocaille** [plausible/analytics#927] [Self-Hosted Support] [answered]
Hello,

first of all congrats on the app!

I am currently trying to install Plausible inside a FreeBSD jail (no docker) from the master branch (but also tried with the 1.3.0.rc.0 tag with the same result). I run the eval for the creation of the Postgres and Clickhouse databases via the command `_build/prod/rel/plausible/bin/plausible eval Plausible.Release.createdb` and then I run into the following issue when trying to do eval the migrations:

```
 ⚡ root@analytics  /usr/local/xiloc/analytics  _build/prod/rel/plausible/bin/plausible eval Plausible.Release.migrate
Loading plausible..
Starting dependencies..
Starting repos..
Running migrations for Elixir.Plausible.Repo
Running migrations for Elixir.Plausible.ClickhouseRepo
** (Clickhousex.Error) Code: 62, e.displayText() = DB::Exception: Syntax error: failed at position 27 ('SAMPLE'): SAMPLE BY user_id
. Expected one of: TTL, COLUMN, ORDER BY, SETTING, QUERY (version 20.7.2.30)

    (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:751: Ecto.Adapters.SQL.raise_sql_call_error/1
    (elixir 1.11.4) lib/enum.ex:1411: Enum."-map/2-lists^map/1-0-"/2
    (ecto_sql 3.5.3) lib/ecto/adapters/sql.ex:838: Ecto.Adapters.SQL.execute_ddl/4
    (ecto_sql 3.5.3) lib/ecto/migration/runner.ex:343: Ecto.Migration.Runner.log_and_execute_ddl/3
    (ecto_sql 3.5.3) lib/ecto/migration/runner.ex:117: anonymous fn/6 in Ecto.Migration.Runner.flush/0
    (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ecto_sql 3.5.3) lib/ecto/migration/runner.ex:116: Ecto.Migration.Runner.flush/0
    (stdlib 3.14.1) timer.erl:166: :timer.tc/1
```

I am not at all familiar with Clickhouse and I guess that would be something related to it...
Any help appreciated! thx in advance

**Comment by ukutaht** [comment, marked as answer]
Hey, thanks.

I haven't documented this yet but version `1.3.0` requires the minimum Clickhouse version to be `20.8.7.15`. Upgrade your CH installation and the migration should run OK.

**Reply by nicocaille** [reply]
ok perfect thanks...I don't know why the FreeBSD ports/pkg version is quite lagging. I'll raise it and in the meantime will try to install a more recent version manually.
